openapi_core/unmarshalling/schemas/unmarshallers.py

```diff
@@ -187,6 +187,23 @@ def _unmarshal_object(self, value=NoValue):
             if properties is None:
                 log.warning("valid oneOf schema not found")
 
+        if self.schema.any_of:
+            properties = None
+            for any_of_schema in self.schema.any_of:
+                try:
+                    unmarshalled = self._unmarshal_properties(
+                        value, any_of_schema)
+                except (UnmarshalError, ValueError):
+                    pass
+                else:
+                    if properties is not None:
+                        log.warning("multiple valid anyOf schemas found")
```

Comment:
it's allowed to have many valid schemas for anyof and properties need to be added rather than replaced